firmware/library/L3_HAL/buzzer.hpp

```diff
@@ -0,0 +1,43 @@
+// Usage:
+// initialize buzzer with pwm pin number
+// BuzzerInterface * buzzer = new Buzzer(1) ;
+// buzzer->Beep(500, 0.9) ; // frequency = 500Hz, Volume = 0.9
+#pragma once
+#include <cstdint>
+#include "L1_Drivers/pwm.hpp" //this?
+
+class BuzzerInterface
+{
+ public:
+    virtual void Initialize() = 0;
+    virtual void Stop() = 0;
+
+    // beep continuously at a certain volume &
+    // at a specified frequency
+    virtual void Beep(uint32_t frequency, float volume) = 0;
+};
+
+class Buzzer : public BuzzerInterface
+{
+ private:
```

Comment:
private section and variables should be at the bottom. See Cpp guidlines in here: https://google.github.io/styleguide/cppguide.html#Class_Format